cli/src/runner.rs

```diff
@@ -179,54 +202,66 @@ pub async fn run_quarantine(
     repo: &BundleRepo,
     delay: std::iter::Take<ExponentialBackoff>,
 ) -> anyhow::Result<QuarantineRunResult> {
-    let quarantine_results = if run_result.failures.is_empty() {
-        QuarantineBulkTestStatus {
-            group_is_quarantined: false,
-            quarantine_results: Vec::new(),
-        }
-    } else {
+    let mut quarantine_results = QuarantineBulkTestStatus {
+        group_is_quarantined: false,
+        quarantine_results: Vec::new(),
+    };
+    let quarantine_config: QuarantineConfig = if !run_result.failures.is_empty() {
         log::info!("Quarantining failed tests");
         let result = Retry::spawn(delay, || {
-            get_quarantine_bulk_test_status(
-                api_address,
-                token,
-                org_url_slug,
-                &repo.repo,
-                &run_result.failures,
-            )
+            get_quarantining_config(api_address, token, org_url_slug, &repo.repo)
         })
         .await;
 
         result.unwrap_or_else(|e| {
             log::error!("Failed to get quarantine results: {:?}", e);
-            QuarantineBulkTestStatus {
-                group_is_quarantined: false,
-                quarantine_results: Vec::new(),
+            QuarantineConfig {
+                is_preview_mode: false,
+                quarantined_tests: Vec::new(),
             }
         })
+    } else {
+        log::info!("No failed tests to quarantine");
+        QuarantineConfig {
+            is_preview_mode: false,
+            quarantined_tests: Vec::new(),
+        }
```

Comment:
nit: similarly here we should use `#[derive(Default)]` on `QuarantineConfig` so we don't have to repeat ourselves and we can refactor this code to be much more simple:

```suggestion
        if let Err(ref err) = result {
            log::error!("Failed to get quarantine results: {:?}", err);
        }

        result.unwrap_or_default()
    } else {
        log::info!("No failed tests to quarantine");
        QuarantineConfig::default()
```